### PR TITLE
feat: Added unit tests for webhook-service and improved error messages

### LINKED ISSUE #5692: Core: Improve error message of webhook-service
## Action Item

The `webhook-service`  should provide a more descriptive message for when a variable cannot be parsed. 

Currently, the message is generic; e.g.:
```
could not parse request 'curl --header 'Content-type: application/json' --request POST --data '{"text":"Hello, World! {{.env.data.project}}"}' https://hooks.slack.com/services/yada_yada_yada'
```

It should report something like:
```
could not parse {{.env.data.project}} for 'curl --header 'Content-type: application/json' --request POST --data '{"text":"Hello, World! {{.env.data.project}}"}' https://hooks.slack.com/services/yada_yada_yada' 
```

## DoD
- [ ] Feature implemented
- [ ] Unit tests

### PR DESCRIPTION
**Pull request by odubajDT**
Signed-off-by: odubajDT <ondrej.dubaj@dynatrace.com>

## This PR
- adds unit tests for webhook-config and improved error messages

### Related Issues

Fixes #5692

### CODE REVIEW

File: webhook-service/handler/handler.go [hunk 1]
```diff
@@ -84,7 +85,7 @@ func (th *TaskHandler) Execute(keptnHandler sdk.IKeptn, event sdk.KeptnEvent) (i
 		}
 		err = keptnHandler.SendFinishedEvent(event, result)
 		if err != nil {
-			return nil, sdkError(fmt.Sprintf("could not send finished event: %s", err.Error()), err)
+			return nil, sdkError(fmt.Sprintf("could not send finished event: request '%s' %s", webhook.Requests[0], err.Error()), err)
```
Comment (bacherfl) [inline webhook-service/handler/handler.go:88]: Can it happen that `webhook.Requests` is empty at this point? Also, is the first of the webhook requests actually tied to this particular error (sending the finished event failed)?

File: webhook-service/handler/handler.go [hunk 2]
```diff
@@ -170,13 +171,13 @@ func (th *TaskHandler) onStartedWebhookExecution(keptnHandler sdk.IKeptn, event
 		// if sendFinished is set, we only need to send one started event
 		// the webhook service will then send a correlating .finished event with the aggregated response payloads
 		if err := keptnHandler.SendStartedEvent(event); err != nil {
-			return sdkError(fmt.Sprintf("could not send .started event: %s", err.Error()), err)
+			return sdkError(fmt.Sprintf("could not send .started event: request '%s': %s", webhook.Requests[0], err.Error()), err)
```
Comment (bacherfl) [inline webhook-service/handler/handler.go:174]: Can it happen that `webhook.Requests` is empty at this point? Also, is the first of the webhook requests actually tied to this particular error (sending the started event failed)?

File: webhook-service/handler/handler.go (hunk 1 above)
Comment (odubajDT) [reply webhook-service/handler/handler.go:88]: Actually it shouldn't happen, we are now checking that webhook.Requests are not empty, see this PR below

File: webhook-service/lib/webhook_config.go [hunk 3]
```diff
@@ -44,5 +48,24 @@ func DecodeWebHookConfigYAML(webhookConfigYaml []byte) (*WebHookConfig, error) {
...
+	for _, webhook := range webHookConfig.Spec.Webhooks {
+		if webhook.Type == "" {
+			return nil, errors.New("Webhook configuration invalid: missing 'webhooks[].Type' part")
+		}
+
+		if webhook.SubscriptionID == "" {
+			return nil, errors.New("Webhook configuration invalid: missing 'webhooks[].SubscriptionID' part")
+		}
+
+		if len(webhook.Requests) == 0 {
```
Comment (odubajDT) [inline webhook-service/lib/webhook_config.go:65]: @bacherfl in this place

File: webhook-service/handler/handler.go (hunk 2 above)
Comment (odubajDT) [reply webhook-service/handler/handler.go:174]: Actually it shouldn't happen, we are now checking that webhook.Requests are not empty, see this PR below

File: webhook-service/handler/handler.go (hunk 1 above)
Comment (bacherfl) [reply webhook-service/handler/handler.go:88]: ok thank you for clarifying :) But still, adding the first of the requests to this error message might be misleading, as the error in this case does not see, to be tied to that particular request

File: webhook-service/lib/template_engine_test.go [hunk 4]
```diff
@@ -42,6 +43,46 @@ func TestTemplateEngine_ParseTemplate(t1 *testing.T) {
 			want:    "",
 			wantErr: true,
 		},
+		{
```
Comment (bacherfl) [inline webhook-service/lib/template_engine_test.go:46]: Is there a way to get any information on what field of a template string was causing the error? I.e. like described in the issue (https://github.com/keptn/keptn/issues/5692)

Comment (odubajDT) [reply webhook-service/lib/template_engine_test.go:46]: will look at it :)

**Comment by odubajDT** [comment]
@bacherfl please re-review? I removed the request[0] as they really do not make sense there and added the required errMsg to the unit tests, hope now it will be good :)

File: webhook-service/lib/template_engine_test.go (hunk 4 above)
Comment (thisthat) [reply webhook-service/lib/template_engine_test.go:50]: Maybe this could help here: https://pkg.go.dev/text/template#Template.Option

Comment (odubajDT) [reply webhook-service/lib/template_engine_test.go:50]: will add, thanks!

Comment (odubajDT) [reply webhook-service/lib/template_engine_test.go:50]: we already have this Option() function used in ParseTemplate(), see https://github.com/keptn/keptn/blob/master/webhook-service/lib/template_engine.go#L16

**Review by bacherfl — APPROVED** [review]

**Pull request merged**